src/middlewared/middlewared/api/base/handler/version.py

```diff
@@ -27,6 +27,22 @@ def __init__(self, version: str, model_name: str):
         super().__init__(f"API version {version!r} does not contain model {model_name!r}")
 
 
+ModelFactory: TypeAlias = Callable[[type[BaseModel]], type[BaseModel]]
+
+
+def _create_model(model_provider: ModelProvider, model_factory: ModelFactory, arg_model_name: str) -> type[BaseModel]:
+    """Call a model factory.
+
+    :param model_provider: An instance of `ModelProvider` that contains the model to pass to `model_factory`.
+    :param model_factory: A callable that returns the model.
+    :param arg_model_name: Name of the model to pass to `model_factory`.
+
+    :raises KeyError: `arg_model_name` is not registered with `model_provider`.
```

Comment:
What if it is an irrelevant `KeyError`? Can an own exception class be used instead? (Or, better, not to use exceptions for business logic).